### PR TITLE
cmake: Use target_link_options/add_link_options

### PR DESCRIPTION
**Pull request by juan-lunarg**

### CODE REVIEW

File: layers/CMakeLists.txt [hunk 1]
```diff
@@ -235,26 +235,12 @@ target_sources(VkLayer_khronos_validation PRIVATE
...
-    set_target_properties(VkLayer_khronos_validation PROPERTIES LINK_FLAGS "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/libVkLayer_khronos_validation.map,-Bsymbolic,--exclude-libs,ALL")
-endif()
-
-# Force generation of the PDB file for Release builds.
-# Note that CMake reduces inlining optimization levels for RelWithDebInfo builds.
-if(MSVC)
-    target_compile_options(VkLayer_khronos_validation PRIVATE "$<$<CONFIG:Release>:/Zi>")
-    # Need to use this instead of target_link_options() for older versions of CMake.
-    target_link_libraries(VkLayer_khronos_validation PRIVATE "$<$<CONFIG:Release>:-DEBUG:FULL>")
+    target_link_options(VkLayer_khronos_validation PRIVATE LINKER:--version-script=${CMAKE_CURRENT_SOURCE_DIR}/libVkLayer_khronos_validation.map,-Bsymbolic,--exclude-libs,ALL)
```
Comment (juan-lunarg) [inline layers/CMakeLists.txt:243]: The `LINKER:` is a new CMake feature intended to help with compiler driver differences:
https://cmake.org/cmake/help/latest/command/target_link_options.html?highlight=target_link_options#handling-compiler-driver-differences

File: layers/CMakeLists.txt [hunk 2]
```diff
@@ -235,26 +235,12 @@ target_sources(VkLayer_khronos_validation PRIVATE
...
 
-if (VVL_ENABLE_ASAN)
-    target_compile_options(VkLayer_khronos_validation PRIVATE -fsanitize=address)
-    # NOTE: Use target_link_options when cmake 3.13 is available on CI
-    target_link_libraries(VkLayer_khronos_validation PRIVATE "-fsanitize=address")
-endif()
-
 if(WIN32)
-    set_target_properties(VkLayer_khronos_validation PROPERTIES LINK_FLAGS "/DEF:\"${CMAKE_CURRENT_SOURCE_DIR}/VkLayer_khronos_validation.def\"")
+    target_link_options(VkLayer_khronos_validation PRIVATE /DEF:${CMAKE_CURRENT_SOURCE_DIR}/VkLayer_khronos_validation.def)
```
Comment (juan-lunarg) [inline layers/CMakeLists.txt:239]: Shouldn't affect aeb96b9f727b945b148b11a944b6eaa953463e3a. FYI to @jeremy-lunarg just in case.

Comment (jeremyg-lunarg) [reply layers/CMakeLists.txt:239]: yeah, I tested this way earlier and it worked for me and my sad sad windows laptop

File: layers/CMakeLists.txt (hunk 1 above)
Comment (ncesario-lunarg) [reply layers/CMakeLists.txt:243]: Thanks for pointing that out!

**Review by ncesario-lunarg — APPROVED** [review]
Hooray! Been wanting this for a long time :)

**Review by jeremyg-lunarg — APPROVED** [review]

**Pull request merged**